packages/web/src/common/store/pages/saved/sagas.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  accountSelectors,
+  APIActivityV2,
+  decodeHashId,
+  encodeHashId,
+  getContext,
+  removeNullable,
+  responseAdapter,
+  savedPageActions as actions,
+  savedPageSelectors,
+  savedPageTracksLineupActions as tracksActions,
+  User,
+  UserTrackMetadata,
+  waitForValue
+} from '@audius/common'
+import { call, fork, put, select, takeLatest } from 'typed-redux-saga'
+
+import { processAndCacheTracks } from 'common/store/cache/tracks/utils'
+import { waitForRead } from 'utils/sagaHelpers'
+
+import tracksSagas from './lineups/sagas'
+const { getSaves } = savedPageSelectors
+const { getAccountUser } = accountSelectors
+
+function* fetchLineupMetadatas(offset: number, limit: number) {
+  const isNativeMobile = yield* getContext('isNativeMobile')
+
+  // Mobile currently uses infinite scroll instead of a virtualized list
+  // so we need to apply the offset & limit
+  if (isNativeMobile) {
+    yield* put(tracksActions.fetchLineupMetadatas(offset, limit))
+  } else {
+    yield* put(tracksActions.fetchLineupMetadatas())
+  }
+}
+
+type LibraryParams = {
+  userId: number
+  offset: number
+  limit: number
+  query: string
+  sortMethod: string
+  sortDirection: string
+}
+
+function* sendLibraryRequest({
+  userId,
+  offset,
+  limit,
+  query,
+  sortMethod,
+  sortDirection
+}: LibraryParams) {
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const { data, signature } = yield* call([
+    audiusBackendInstance,
+    audiusBackendInstance.signDiscoveryNodeRequest
+  ])
+  const audiusSdk = yield* getContext('audiusSdk')
+  const sdk = yield* call(audiusSdk)
+
+  const savedTracksResponse = (yield* call(
+    [sdk.full.users, sdk.full.users.getUserLibraryTracks as any],
+    {
+      id: encodeHashId(userId),
+      userId: encodeHashId(userId),
+      offset,
+      limit,
+      query,
+      sortMethod,
+      sortDirection,
+      type: 'all',
+      encodedDataMessage: data,
+      encodedDataSignature: signature
+    }
+  )) as any
+  const savedTracksResponseData = savedTracksResponse.data as APIActivityV2[]
```

Comment:
Bummer that we have to cast this. Would be a great use case for `satisfies` when our codebase supports it. Btw, I think `APIActivityV2` doesn't exist?